sdk-cpp/include/do_download_property.h

```diff
@@ -42,56 +46,58 @@ enum class download_property
     integrity_check_info,           // std::string
     integrity_check_mandatory,      // boolean
     total_size_bytes,               // uint64
-    // Fe and beyond
+
+    // For the COM interface, the following properties are available only in Windows 21H1 and beyond
     disallow_on_cellular,           // bool
     http_custom_auth_headers,       // std::string
 };
 
 class download_property_value
 {
 
+// CDownloadImpl is declared as a friend class because it needs to access the platform-specific native value for download_property_value
+// The type of the native value is defined in CDownloadPropertyValueInternal, because DO header files are platform agnostic
+// This is so any user of the SDK does not have to worry about supplying platform specific compile definitions to use the SDK
+friend class details::CDownloadImpl; 
+
 public:
-    using native_type = VARIANT;
     using status_callback_t = std::function<void(download&, download_status&)>;
-
     download_property_value() = default;
 
+    /* User Note:
+    These constructors will throw on failure, unless DO_DISABLE_EXCEPTIONS is toggled, in which case init_code() must be called to check if initialization failed
+    */
     explicit download_property_value(const std::string& val);
-    explicit download_property_value(UINT val);
-    explicit download_property_value(UINT64 val);
+    explicit download_property_value(uint32_t val);
+    explicit download_property_value(uint64_t val);
     explicit download_property_value(bool val);
     explicit download_property_value(std::vector<unsigned char>& val); // DODownloadProperty_SecurityContext
 
     explicit download_property_value(const status_callback_t& val);
```

Comment:
This was the other iffy thing to me - it feels weird to compile the SDK in a way such that these constructors will have different behavior. But it is definitely a working solution - just not sure if it's the best one.

Reply:
The standard convention here is to provide make() methods and a default constructor that is private.

int32_t make(const std::string& val);
int32_t make(uint32_t val);
...

Reply:
But isn't this a little awkward for consumers who will want to use a throwing interface? They might expact constructors utilizing RAII and throwing on failure. Should we expect people using both the throwing/non-throwing interface to use make() methods?

Reply:
We can offer both throwing and non throwing versions of make methods too.

Think of it as this way - constructors should no longer do anything that can fail (except allocating memory/other objects of course). If anything more is required, for example other initialization steps as in the CDownloadImpl example, then we should make the constructors private and offer 'builder' methods (the make methods) which can have both throwing and non-throwing versions.